packages/select/src/components/select/suggest.tsx

```diff
@@ -102,6 +103,13 @@ export class Suggest<T> extends React.PureComponent<ISuggestProps<T>, ISuggestSt
         );
     }
 
+    public componentWillReceiveProps(nextProps: ISuggestProps<T>) {
+        // If the selected item prop changes, update the underlying state.
+        if (nextProps.selectedItem !== this.state.selectedItem) {
```

Comment:
Maybe it should be better to use isEqual from lodash instead of the simple operator, what do you think?

Reply:
❌ not adding a dependency on lodash. this issue is common to all the select components and i think should be solved by an optional `itemEquals` prop a la `itemDisabled`

Reply:
but not in this PR 👍